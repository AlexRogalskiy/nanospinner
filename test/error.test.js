import { createSpinner } from '../index.js'

let stdout = { out: '' }
stdout.write = symbols => {
  stdout.out += symbols
}

it('marks spinner as error', () => {
  let spinner = createSpinner('#error', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.error()

  expect(stdout.out).toMatchSnapshot()
})

it('marks spinner as error with message', () => {
  let spinner = createSpinner('#error', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.success({ text: 'Error\n' })

  expect(stdout.out).toMatchSnapshot()
})

it('marks spinner as error with mark', () => {
  let spinner = createSpinner('#error', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.success({ mark: '!' })

  expect(stdout.out).toMatchSnapshot()
})
