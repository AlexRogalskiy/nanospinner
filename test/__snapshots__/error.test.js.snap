// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marks spinner as error 1`] = `
"[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[31mâœ–[39m #error
"
`;

exports[`marks spinner as error with mark 1`] = `
"[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[31mâœ–[39m #error
[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[32mâœ”[39m Error

[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[32mâœ”[39m #error
"
`;

exports[`marks spinner as error with message 1`] = `
"[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[31mâœ–[39m #error
[1G[33mâ ‹[39m #error[1G[33mâ ™[39m #error[1G[33mâ ¹[39m #error[2K[1G[32mâœ”[39m Error

"
`;
